internal/auth/mux.go

```diff
@@ -17,34 +16,33 @@ type AuthenticatorMux struct {
 	authenticators []*Authenticator
 }
 
-func NewAuthenticatorMux(opts *Options, statsdClient *statsd.Client) (*AuthenticatorMux, error) {
+func NewAuthenticatorMux(config Configuration, statsdClient *statsd.Client) (*AuthenticatorMux, error) {
 	logger := log.NewLogEntry()
 
 	var validator func(string) bool
-	if len(opts.EmailAddresses) != 0 {
-		validator = options.NewEmailAddressValidator(opts.EmailAddresses)
+	if len(config.AuthorizeConfig.EmailConfig.Addresses) != 0 {
+		validator = options.NewEmailAddressValidator(config.AuthorizeConfig.EmailConfig.Addresses)
 	} else {
-		validator = options.NewEmailDomainValidator(opts.EmailDomains)
+		validator = options.NewEmailDomainValidator(config.AuthorizeConfig.EmailConfig.Domains)
 	}
 
-	// one day, we will contruct more providers here
-	idp, err := newProvider(opts)
-	if err != nil {
-		logger.Error(err, "error creating new Identity Provider")
-		return nil, err
-	}
-	identityProviders := []providers.Provider{idp}
 	authenticators := []*Authenticator{}
-
 	idpMux := http.NewServeMux()
-	for _, idp := range identityProviders {
+
+	for slug, providerConfig := range config.ProviderConfigs {
+		idp, err := newProvider(providerConfig, config.SessionConfig)
+		if err != nil {
+			logger.Error(err, fmt.Sprintf("error creating provider.%s", slug))
+			return nil, err
+		}
+
 		idpSlug := idp.Data().ProviderSlug
-		authenticator, err := NewAuthenticator(opts,
+		authenticator, err := NewAuthenticator(config,
 			SetValidator(validator),
 			SetProvider(idp),
-			SetCookieStore(opts, idpSlug),
+			SetCookieStore(config, idpSlug),
```

Comment:
Should we be even more explicit here with `SetCookieStore` and pass in `config.SessionConfig` instead? It doesn't look like it needs to access any other config data structures